src/schema/schema.c

```diff
@@ -117,28 +35,26 @@ Index* Schema_GetIndex(Schema *s, const char *attribute) {
     return NULL;
 }
 
-void Schema_AddIndex(Schema *s, char *attribute, Index *idx) {    
-    // Make sure attribute is part of schema.
-    Attribute_ID id = Schema_GetAttributeID(s, attribute);
-    assert(id != ATTRIBUTE_NOTFOUND);
-
+int Schema_AddIndex(Schema *s, char *attribute, Attribute_ID attr_id) {
     // Make sure attribute isn't already indexed.
-    assert(Schema_GetIndex(s, attribute) == NULL);
+    if (Schema_GetIndex(s, attr_id) != NULL) return INDEX_FAIL;
+
+    // Populate an index for the label-attribute pair using the Graph interfaces.
+    GraphContext *gc = GraphContext_GetFromTLS();
```

Comment:
Pass GraphContext as an argument

Reply:
That would require `graphcontext.h` to be included in the header file, which causes some ugly circular referencing issues.